neuralpp/symbolic/basic_expression.py

```diff
@@ -281,10 +289,14 @@ def basic_add_operation(type_) -> BasicAbelianOperation:
     return BasicAbelianOperation(operator.add, BasicConstant(0, type_), operator.neg)
 
 
-class BasicSummation(BasicQuantifierExpression):
-    def __init__(self, type_: type, index: Variable, constraint: Context, body: Expression):
-        """
-        Expect type_ to be the argument type/return type of the summation.
-        E.g., the type_ of Summation{i in [0,100]}(i) should be `int`.
-        """
-        super().__init__(basic_add_operation(type_), index, constraint, body)
+def BasicSummation(type_: type, index: Variable, constraint: Context, body: Expression) -> BasicQuantifierExpression:
+    """
+    Expect type_ to be the argument type/return type of the summation.
+    E.g., the type_ of Summation{i in [0,100]}(i) should be `int`.
+    """
+    return BasicQuantifierExpression(basic_add_operation(type_), index, constraint, body, False)
+
+
+def BasicIntegral(index: Variable, constraint: Context, body: Expression) -> BasicQuantifierExpression:
```

Comment:
N00b question: why are we making `BasicSummation` and `BasicIntegral` functions instead of having them as classes now?

Reply:
I should have explained this in the comment. Thanks for asking.

Because we're constantly re-creating new `BasicQuantifierExpression` by `new_quantifier_expression` (called by `replace()` which is used everywhere in simplification and normalization). And these new `BasicQuantifierExpression`s are not `BasicSummation` or `BasicIntegral`. So it's kinda weird to me that we have two notions (`BasicSummation` and `BasicQuantifierExpression` whose operation is `operator.add`) at the same time representing the same thing, and sometimes the former get transformed into the latter. So I just decided to make `BasicSummation` and `BasicIntegral` functions.

Reply:
Gotcha, thanks for the clarification! I was asking because I felt it would be very convenient to use the class to check if an expression is integral, but I agree that it can be tricky to deal with summations and integrals that have the base quantifier type.